CHANGELOG.md

```diff
@@ -8,6 +8,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 
 ### Breaking changes
+* If `jj squash` receives paths or `--interactive`, then it will not abandon the
+  source commit even if it became empty.
```

Comment:
nit: I think people usually add new items last, unless there's a reason it belongs first (because it's more important than other changes or because it's related to the second item)